### PR TITLE
[FIX] Doxygen warnings in Alignment Module

### PR DESCRIPTION
**Pull request by joergi-w**

### CODE REVIEW

File: include/seqan3/alignment/configuration/align_config_mode.hpp [hunk 1]
```diff
@@ -17,7 +17,11 @@
 #include <seqan3/alignment/configuration/detail.hpp>
 #include <seqan3/core/algorithm/pipeable_config_element.hpp>
 
-namespace seqan3::detail
+namespace seqan3
+{
+
+// We need to nest the namespace explicitly, such that doxygen is capable of correct parsing.
```
Comment (eseiler) [inline include/seqan3/alignment/configuration/align_config_mode.hpp]: I'd remove the comment from here since it will affect every nested namespace.

**Comment by eseiler** [comment]
No more nested namespaces @seqan/core 🙈 
https://github.com/doxygen/doxygen/issues/6765

**Comment by h-2** [comment]
> No more nested namespaces @seqan/core see_no_evil

Yep, I was aware of the issure already. We definitely need to wait for an update of doxygen before switching from .14

File: include/seqan3/alignment/configuration/align_config_mode.hpp (hunk 1 above)
Comment (joergi-w) [reply include/seqan3/alignment/configuration/align_config_mode.hpp]: Fine, then I remove the comments.

**Comment by h-2** [comment]
Please do not change the namespaces. This is not something we discussed and I am strongly against changing wide-ranging things like this just because of a bug. The other changes were errors on our side (missing documentation, double relates), but the namespaces-thing is something that needs to be fixed in doxygen.

**Comment by joergi-w** [comment]
> Please do not change the namespaces. 

@rrahn @smehringer 
Do you agree that I should revert the namespace nesting?
If I do, the problem is that there are large chains of follow-up errors, e.g. some functions go to `seqan3::seqan3::` namespace and produce a bunch of unresolved references. Therefore, it becomes really hard to distinguish them from real documentation errors that should be fixed.

How likely will the doxygen routine be repaired **before** our release?

**Comment by joergi-w** [comment]
As discussed I reverted the namespace nesting. This PR is ready for a final review.

File: include/seqan3/alignment/configuration/align_config_aligned_ends.hpp [hunk 2]
```diff
@@ -398,11 +400,13 @@ class end_gaps
...
 
 /*!\name Type deduction guides
- * \relates seqan3::end_gaps
  * \{
  */
 
-//!\brief Deduces the end-gap specifier from the constructor arguments.
+/*!\brief Deduces the end-gap specifier from the constructor arguments.
+ * \relates seqan3::end_gaps
+ * \tparam ends_t A parameter pack containing at most 4 sequence end-gap specifiers.
```
Comment (rrahn) [inline include/seqan3/alignment/configuration/align_config_aligned_ends.hpp]:
```suggestion
 * \tparam ends_t A template parameter pack containing at most 4 sequence end-gap specifiers.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 3]
```diff
@@ -77,8 +77,10 @@ class advanceable_alignment_coordinate
 {
 public:
 
-    //!\name Member types
-    //!\{
+    /*!\name Member types
+     * \{
+     */
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]: does this trigger warnings in doxygen?

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 4]
```diff
@@ -160,6 +166,9 @@ class advanceable_alignment_coordinate
         return *this;
     }
 
+    /*!\brief Post-increments the coordinate depending on the set policy by one.
+     * \return a copy of *this that was made before the change
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]:
```suggestion
     * \return a seqan3::detail::advanceable_alignment_coordinate that holds an unchanged value.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 5]
```diff
@@ -182,6 +194,9 @@ class advanceable_alignment_coordinate
         return *this;
     }
 
+    /*!\brief Post-decrements the coordinate depending on the set policy by one.
+     * \return a copy of *this that was made before the change
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]:
```suggestion
     * \return a seqan3::detail::advanceable_alignment_coordinate that holds an unchanged value.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 6]
```diff
@@ -192,6 +207,10 @@ class advanceable_alignment_coordinate
         return tmp;
     }
 
+    /*!\brief Returns the coordinate which is advanced depending on the set policy by offset.
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]:
```suggestion
    /*!\brief Returns the coordinate which is advanced depending on the set policy by `offset`.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 7]
```diff
@@ -204,6 +223,10 @@ class advanceable_alignment_coordinate
         return *this;
     }
 
+    /*!\brief Returns the coordinate which is advanced depending on the set policy by -offset.
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]:
```suggestion
    /*!\brief Returns the coordinate which is advanced depending on the set policy by `-offset`.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 8]
```diff
@@ -216,6 +239,10 @@ class advanceable_alignment_coordinate
         return *this;
     }
 
+    /*!\brief Returns a new coordinate which is advanced depending on the set policy by offset.
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]:
```suggestion
    /*!\brief Returns a new coordinate which is advanced depending on the set policy by `offset`.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 9]
```diff
@@ -216,6 +239,10 @@ class advanceable_alignment_coordinate
...
+     * \param offset The value to add to the coordinate.
+     * \return a copy of *this advanced by offset
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]:
```suggestion
     * \return a seqan3::detail::advanceable_alignment_coordinate that holds the updated value.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 10]
```diff
@@ -226,6 +253,10 @@ class advanceable_alignment_coordinate
         return tmp;
     }
 
+    /*!\brief Returns a new coordinate which is advanced depending on the set policy by -offset.
+     * \param offset The value to subtract from the coordinate.
+     * \return a copy of *this advanced by -offset
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]:
```suggestion
     * \return a seqan3::detail::advanceable_alignment_coordinate that holds an unchanged value.
```

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp [hunk 11]
```diff
@@ -236,6 +267,7 @@ class advanceable_alignment_coordinate
         return tmp;
     }
 
+    //!\brief Returns the difference of this and another coordinate depending on the set policy.
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/alignment_coordinate.hpp]: missing `\param other`?

File: include/seqan3/alignment/pairwise/alignment_result.hpp [hunk 12]
```diff
@@ -119,8 +119,9 @@ class alignment_result
 public:
     /*!\name Constructors, destructor and assignment
      * \{
-     *
-     * \brief Constructor to pass the alignment result traits.
+     */
+
+    /*!\brief Constructor to pass the alignment result traits.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/alignment_result.hpp]:
```suggestion
    /*!\brief Constructs a seqan3::alignment_result from a `alignment_result_traits` object.
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 13]
```diff
@@ -77,12 +81,19 @@ class alignment_range
...
+         * \return *this
+         */
         iterator_type & operator++(/*pre*/) noexcept
         {
             range_ptr->next();
             return *this;
         }
 
+        /*!\brief Post-increments the iterator by one.
+         * \return a copy of *this that was made before the change
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
         * \return an `iterator_type` that holds an unchanged value.
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 14]
```diff
@@ -93,22 +104,26 @@ class alignment_range
          * \{
          */
 
+        //!\brief Test equality.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
        //!\brief Checks whether `*this` is equal to the sentinel.
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 15]
```diff
@@ -93,22 +104,26 @@ class alignment_range
...
         }
 
+        //!\brief Test equality.
         friend constexpr bool operator==(std::ranges::default_sentinel_t const & lhs,
                                          iterator_type const & rhs) noexcept
         {
             return rhs == lhs;
         }
 
+        //!\brief Test inequality.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
        //!\brief Checks whether `*this` is not equal to the sentinel.
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 16]
```diff
@@ -93,22 +104,26 @@ class alignment_range
          * \{
          */
 
+        //!\brief Test equality.
         constexpr bool operator==(std::ranges::default_sentinel_t const &) const noexcept
         {
             return range_ptr->eof();
         }
 
+        //!\brief Test equality.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
        //!\brief Checks whether `lhs` is equal to `rhs`
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 17]
```diff
@@ -93,22 +104,26 @@ class alignment_range
...
             return rhs == lhs;
         }
 
+        //!\brief Test inequality.
         constexpr bool operator!=(std::ranges::default_sentinel_t const & rhs) const noexcept
         {
             return !(*this == rhs);
         }
 
+        //!\brief Test inequality.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
        //!\brief Checks whether `lhs` is not equal to `rhs`.
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 18]
```diff
@@ -170,6 +185,10 @@ class alignment_range
     /*!\name Iterators
      * \{
      */
+
+    /*!\brief Returns an iterator to the first element of the alignment range.
+     * \return An iterator to the beginning of the alignment range.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
     * \return An iterator to first element.
     *
     * \details
     *
     * Invocation of this function will trigger the computation of the first alignment. 
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 19]
```diff
@@ -180,6 +199,9 @@ class alignment_range
     const_iterator begin() const = delete;
     const_iterator cbegin() const = delete;
 
+    /*!\brief Returns an iterator to the end (i.e. the element after the last element) of the alignment range.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
    /*!\brief Returns a sentinel signaling the end of the alignment range.
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 20]
```diff
@@ -180,6 +199,9 @@ class alignment_range
...
+     * \return An iterator to the element following the last valid element of the alignment range.
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
     * \return a sentinel.
```

File: include/seqan3/alignment/pairwise/execution/alignment_range.hpp [hunk 21]
```diff
@@ -180,6 +199,9 @@ class alignment_range
...
+     */
```
Comment (rrahn) [inline include/seqan3/alignment/pairwise/execution/alignment_range.hpp]:
```suggestion
     *
     * \details 
     * 
     * The alignment range is an input range and the end is reached when the internal buffer over the alignment results has signaled end-of-stream.
     */
```

File: include/seqan3/alignment/scoring/gap_scheme.hpp [hunk 22]
```diff
@@ -208,13 +212,17 @@ class gap_scheme
     }
     //!\}
 
-    //!\name Comparison operators
-    //!\{
+    /*!\name Comparison operators
+     * \{
+     */
```
Comment (rrahn) [inline include/seqan3/alignment/scoring/gap_scheme.hpp]: Why these changes?

File: include/seqan3/alignment/scoring/gap_scheme.hpp [hunk 23]
```diff
@@ -208,13 +212,17 @@ class gap_scheme
...
+
+    //!\brief Test equality.
```
Comment (rrahn) [inline include/seqan3/alignment/scoring/gap_scheme.hpp]: see comments before.

File: include/seqan3/alignment/scoring/scoring_scheme_base.hpp [hunk 24]
```diff
@@ -220,11 +224,14 @@ class scoring_scheme_base
 
     //!\name Comparison operators
     //!\{
+
+    //!\brief Test equality.
```
Comment (rrahn) [inline include/seqan3/alignment/scoring/scoring_scheme_base.hpp]: see comments before

File: include/seqan3/alignment/matrix/alignment_coordinate.hpp (hunk 3 above)
Comment (joergi-w) [reply include/seqan3/alignment/matrix/alignment_coordinate.hpp]: Sadly, yes. It has to be a block comment, otherwise the following code remains undocumented:
```
seqan3/alignment/matrix/alignment_coordinate.hpp:83: warning: Member difference_type (typedef) of class seqan3::detail::advanceable_alignment_coordinate is not documented.
```

Comment (rrahn) [reply include/seqan3/alignment/matrix/alignment_coordinate.hpp]: 😭

File: include/seqan3/alignment/scoring/gap_scheme.hpp (hunk 22 above)
Comment (joergi-w) [reply include/seqan3/alignment/scoring/gap_scheme.hpp]: see above

**Comment by joergi-w** [comment]
Rebased  on master, because there were changes in the alignment module.
